File: identity.js
function setIdetityHeader(r) {
    var authType = r.headersIn['x-auth-type'] || '';
    var orgId = r.headersIn['x-org-id'] || r.variables.env_org_id;

    var certTemplate = JSON.parse(r.variables.cert_template);
    var jwtTemplate = JSON.parse(r.variables.jwt_template);

    var identity;

    if (authType === 'cert-auth') {
        identity = JSON.stringify({
            auth_type: certTemplate.auth_type,
            cn: r.headersIn['x-cn'] || 'unknown',
            org_id: orgId || 'unknown'
        });
    } else if (authType === 'jwt-auth') {
        var userData = {};
        try {
            userData = JSON.parse(r.headersIn['x-user'] || '{}');
        } catch (e) {
            r.warn('Failed to parse x-user header: ' + e);
        }
        identity = JSON.stringify({
            auth_type: jstTemplate.auth_type,
            username: userData.username || 'unknown',
            email: userData.email || 'unknown',
            first_name: userData.first_name || 'unknown',
            last_name: userData.last_name || 'unknown',
            org_id: orgId || 'unknown'            
        });
    } else {
        identity = '{}';
    }

    r.headersOut['x-rh-identity'] = identity;
    return identity;
}

export default { setIdetityHeader };

File: identity.pl
use JSON;
use MIME::Base64;

sub set_identity_header {
    my $r = shift;

    my $auth_type = $r->header_in("x-auth-type") || "";
    my $org_id = $r->header_in("x-org-id") || $r->variable("env_org_id");

    my $cert_template = decode_json($r->variable("cert_template"));
    my $jwt_template =decode_json($r->variable("jwt_template"));

    my $identity;
    my $identity_b64;

    if($auth_type eq "cert-auth") {
        $cert_template->{cn} = $r->header_in("x-cn") || "unknown";
        $cert_template->{org_id} = $org_id || "unknown";
        $identity = encode_json($cert_template);
    }
    elsif ($auth_type eq "jwt-auth") {
        my $user_json = r->header_in("x-user") || "{}";
        my $user_data = decode_json($user_json);

        $jwt_template->{username} = $user_data->{username} || "unknown";
        $jwt_template->{email} = $user_data->{email} || "unknown";
        $jwt_template->{first_name} = $user_data->{first_name} || "unknown";
        $jwt_template->{last_name} = $user_data->{last_name} || "unknown";
        $jwt_template->{org_id} = $org_id || "unknown";
        $identity = encode_json($jwt_template);
    }
    else {
        $identity = "{}";
    }

    $identity_b64 = encode_base64($identity);
    r->header_out("x-rh-identity", $identity_b64);
    return identity_b64;
}

1;
